### ISSUE TITLE
Cannot broadcast 

### ISSUE THREAD

**Issue by abduramann** [secretanalytics/secret-sdk-python#6]
When I try to broadcast, I get this message:

secret_sdk.exceptions.LCDResponseError: Status 400 - this transaction cannot be broadcasted via legacy REST endpoints, because it does not support Amino serialization. Please either use CLI, gRPC, gRPC-gateway, or directly query the Tendermint RPC endpoint to broadcast this transaction. The new REST endpoint (via gRPC-gateway) is POST /cosmos/tx/v1beta1/txs. Please also see the REST endpoints migration guide at https://docs.cosmos.network/master/migrations/rest.html for more info                                                                                                                                                                                                                                      

Then I've updated /txs endpoint  as /secret/tx/v1beta1/txs. This time I got 501 since the format of message also changed. 

So, forking didn't work. I think it will require more work than I think :) Will you update SDK soon? Is there any ETA?

Thanks.

**Comment by SecretShell** [comment]
What type of transaction are you trying to make?

**Comment by paul-HDD** [comment]
This is most likely an error in the message payload of your tx, could you give more details on what you're trying to broadcast ?

**Comment by abduramann** [comment]
While smart contract execution:

```
MsgExecuteContract(
    "ADDRESS",
    "TOKEN_CONTRACT",
    {
      "send": {
        "contract": "PAIR_CONTRACT",
        "amount": str(round(1 * 1000000)),
        "msg": str(base64.b64encode(('{
          "swap": {
            "belief_price": "1",
            "max_spread": "0.005",
            "to": "",
          }
        }').encode()), "utf-8")
      }
    }
)

```

As I wrote in Telegram group, the problem is about the wrong endpoint, the wrong message content, and some issues I didn't know. 

Wrong endpoint:  /txs --> /secret/tx/v1beta1/txs
Wrong content: "tx" --> "tx_bytes", mode: block --> "BROADCAST_MODE_BLOCK"

I've updated _broadcast() method in https://github.com/secretanalytics/secret-sdk-python/blob/main/secret_sdk/client/lcd/api/tx.py according to what I wrote above. I'm getting a 501 error without any message.

As a quick solution, if I replace all endpoints of the latest https://github.com/terra-money/terra.py (in which these issues are fixed) with Secret's ones, would it work? Is it worth trying?

**Comment by abduramann** [comment]
I've found the issue. What I wrote above is incorrect. I still couldn't complete the whole process since tx is failed to execute but at least my tx is added to the block successfully.

While I was searching for an easy solution to this, I saw the solution in https://githubmemory.com/@tjpeterson94. I'm not sure whether the problem is the same but it has sparked an idea in my mind. 

Following the example in https://github.com/secretanalytics/secret-sdk-python/blob/main/docs/guides/smart_contracts.rst, I was using these lines:

```
execute_tx = test1.create_and_sign_tx(
    msgs=[execute], fee=StdFee(1000000, Coins(uscrt=1000000))
)

execute_tx_result = secret.tx.broadcast(execute_tx)
```
and I got the message in my first post.

Now, instead, I use the line below and my tx is broadcasted successfully:

`execute_tx = wallet.execute_tx(msg)`
